.github/workflows/testing.yml

```diff
@@ -0,0 +1,217 @@
+name: Automated Tests
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python_version: [2.7, 3.5, 3.6, 3.7, 3.8]
+        exclude:
+          # do not test each porsisbl (os, python version) configuration
+          - os: windows-latest
+            python_version: 3.5
+          - os: windows-latest
+            python_version: 3.6
+          - os: windows-latest
+            python_version: 3.8
+          - os: macos-latest
+            python_version: 3.5
+          - os: macos-latest
+            python_version: 3.6
+          - os: macos-latest
+            python_version: 3.8
```

Comment:
Why are those excluded? In particular I think we should test the latest python version (3.8) for all the supported platforms.

Skipping 3.6 or 3.7 on some platforms to speed up CI is fine but I would rather always test the boundaries (2.7, 3.5 and 3.8 at the moment) on all platforms.

Reply:
Yes I just mimicated the entries we had for `windows`. You are right, we might as well test the boundaries.